bundle_gems

```diff
@@ -69,12 +60,56 @@ if gem_file_lock.empty?
 end
 @logger.info "Using #{gem_file_lock}"
 
+# Get the bundler version defined in gem_file_lock...
+next_line = false
+File.readlines(gem_file_lock).each do |line|
+  @bundledwith = line if next_line
+  next_line = false if next_line
+  next_line = true if line.start_with?('BUNDLED WITH')
+end
+@bundledwith = @bundledwith.lstrip.chomp if @bundledwith
+
 def run_command(command:, environment: {})
   stdout_and_stderr_str, status = Open3.capture2e(environment, command)
   @logger.info stdout_and_stderr_str
   abort(stdout_and_stderr_str) unless status.success?
 end
 
+def bundle(command:)
+  if @bundler_directory
+    bundler_bin = "#{@bundler_directory}/bin/bundle _#{Bundler::VERSION}_"
+    run_command(command: "#{bundler_bin} #{command}", environment: { 'GEM_HOME' => @bundler_directory })
+  else
+    run_command(command: "bundle #{command}")
+  end
+end
+
+def install_bundler(version:)
+  @logger.info "Installing bundler version #{version}"
+  bundler_directory = Dir.mktmpdir(nil, Dir.pwd)
+
+  Dir.chdir(bundler_directory) do
+    run_command(command: "gem install bundler -v #{version}", environment: { 'GEM_HOME' => bundler_directory })
+    # adapt LOAD_PATH, gosh this is hacky...
+    bundler_lib_path = Dir.glob('gems/bundler-*/lib').first
+    $LOAD_PATH.unshift("#{bundler_directory}/#{bundler_lib_path}")
+    require 'bundler'
+  end
+
+  ENV['GEM_HOME'] = bundler_directory
+
+  bundler_directory
+end
+
+# Install and require the right bundler version
+if @bundledwith
+  @bundler_directory = install_bundler(version: @bundledwith)
+  @logger.info "Using bundler (#{Bundler::VERSION}) defined by 'BUNDLED WITH' in #{gem_file_lock}..."
+else
+  require 'bundler'
+  @logger.info "Using system bundler (#{Bundler::VERSION}), no 'BUNDLED WITH' in #{gem_file_lock}..."
+end
+
 # Execute with cpio strategy
 if options[:strategy] == :cpio
   FileUtils.cp gem_file, File.join(outdir, 'Gemfile')
```

Comment:
@hennevogel I just tested this a bit locally, I realized it fails in this line when no `--outdir` is set, because it tries to overwrite the existing `Gemfile` and `Gemfile.lock`

Reply:
you mean inside the git checkout? Yeah to be expected I would say. There is `outdir = options[:outdir] || Dir.pwd`